test/sensor/camera/test_camera.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import torch
+
+from kornia.geometry.vector import Vector3
+from kornia.image import ImageSize
+from kornia.sensor.camera import CameraModel, CameraModelType
+from kornia.testing import BaseTester
+
+
+class TestPinholeCamera(BaseTester):
+    def _make_rand_data(self, batch_size, device, dtype):
+        params = torch.rand(batch_size, 4).to(dtype).to(device)
+        image_sizes = torch.randint(1, 100, (batch_size, 2)).to(dtype).to(device)
+        return params, ImageSize(image_sizes[:, 0], image_sizes[:, 1])
+
+    def test_smoke(self, device, dtype):
+        params, image_size = self._make_rand_data(1, device, dtype)
+        cam = CameraModel(image_size, CameraModelType.PINHOLE, params)
+        assert isinstance(cam, CameraModel)
+        self.assert_close(cam.params, params)
+        self.assert_close(image_size.height, cam.height)
+        self.assert_close(image_size.width, cam.width)
+
+    def test_cardinality(self, device, dtype):
+        pass
+
+    def test_exception(self, device, dtype):
+        # test for invalid params for different camera models
+        params = torch.tensor([1.0, 1.0, 1.0, 1.0, 1.0])
+        image_size = ImageSize(100, 100)
+        with pytest.raises(ValueError):
+            CameraModel(image_size, CameraModelType.PINHOLE, params)
+        with pytest.raises(ValueError):
+            CameraModel(image_size, CameraModelType.BROWN_CONRADY, params)
+        with pytest.raises(ValueError):
+            CameraModel(image_size, CameraModelType.KANNALA_BRANDT_K3, params)
+        with pytest.raises(ValueError):
+            CameraModel(image_size, CameraModelType.ORTHOGRAPHIC, params)
+
+    def test_gradcheck(self, device):
+        pass
```

Comment:
skip if is unnecessary

Reply:
keep it as is? aren't these abstract methods?